src/Parser/RestoDruid/Modules/Traits/BlessingOfTheWorldTree.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import calculateEffectiveHealing from 'Parser/Core/calculateEffectiveHealing';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+const BLESSING_OF_THE_WORLD_TREE_HEALING_INCREASE = 0.03;
+
+/**
+ * Blessing of the World Tree (Artifact Trait)
+ * Increases the healing done by Healing Touch and initial healing done by Regrowth by 3%
+ */
+class BlessingOfTheWorldTree extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  rank = 0;
+  healing = 0;
+
+  on_initialized() {
+    this.rank = this.combatants.selected.traitsBySpellId[SPELLS.BLESSING_OF_THE_WORLD_TREE.id];
+    this.active = this.rank > 0;
+  }
+
+  on_byPlayer_heal(event) {
+    if(event.ability.guid !== SPELLS.HEALING_TOUCH.id || (event.ability.guid !== SPELLS.REGROWTH.id && event.tick) ) {
+      return;
+    }
+
+    this.healing += calculateEffectiveHealing(event, BLESSING_OF_THE_WORLD_TREE_HEALING_INCREASE * this.rank)/ this.rank;
+  }
```

Comment:
Stylistically I think it's better to use an affirmative 'if' instead of using an early return, like:

```
if(event.ability.guid === SPELLS.HEALING_TOUCH.id || (event.ability.guid === SPELLS.REGROWTH.id && !event.tick) ) {
  +    this.healing += calculateEffectiveHealing(event, BLESSING_OF_THE_WORLD_TREE_HEALING_INCREASE * this.rank)/ this.rank;
}
```

I think this mostly doesn't matter, just thought I'd throw it out there.

Reply:
I believe @MartijnHols mentioned somewhere (?) that he enforces early returns.

Reply:
I'll ask him what the reason is for that. I always preferred early returning only for exceptional behavior. When implementing pseudocode of "if X, then do A", I prefer:
```
if(X) {
   do A;
}
```
rather than:
```
if(!X) {
   return;
}
do A;
```

Reply:
I believe it's to avoid potential nested if-cases. This is probably only applicable once you start nesting over a depth of 3.
https://3layer.com.br/sonar/rules/show/javascript:NestedIfDepth?layout=false

```
if(X) {
   some code;
   if(Y) {
      some code;
      if(Z) {
           do A;
      }
   }
}
```

```
if(!X) {
   return;
}

some code;
if(!Y) {
   return;
}
some code;
if(!Z) {
   return;
}

do A
```

Reply:
Early returns are preferred Early returns are preferred when they improve readability and maintainability. If they don't then don't use them. It's more a guideline as it's near impossible to make rules that apply just as well everywhere.

I find nested if-statements harder to follow, change, add onto and remove as they require you to understand every single layer before you can modify the layers. On the other hand early returning keeps your method 1 layer deep and acts more like exceptions to the regular flow which is a lot easier to track, understand and modify. I also reckon often when you need multiple nested if-statements in one function you might need to start refactoring.

In this instance either could work, because of the specific Regrowth tick filtering I reckon it's more readable to nest the expression.